// mailtrap-sdk/dist/src/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MailtrapSDK = void 0;
const axios_1 = __importDefault(require("axios"));
const config_1 = __importDefault(require("./lib/config"));
const errors_util_1 = require("./lib/utils/errors.util");
const flatten_email_util_1 = require("./lib/utils/flatten-email.util");
const { MAILTRAP } = config_1.default;
const { API_BASE, ENDPOINTS } = MAILTRAP;
const { MailtrapAPIError } = errors_util_1.ERRORS;
/**
 * Mailtrap SDK for sending email over Mailtrap Testing API.
 */
class MailtrapSDK {
    MailtrapInstance;
    testInboxId;
    constructor(params) {
        this.MailtrapInstance = axios_1.default.create({
            baseURL: API_BASE,
            headers: {
                'Authorization': `Bearer ${params.accessToken}`
            }
        });
        this.testInboxId = params.testInboxId;
    }
    /**
     * Sends email with given `params` which include recipient, sender, subject, text, etc.
     */
    async send(params) {
        const data = (0, flatten_email_util_1.flattenEmailData)(params);
        const endpoint = ENDPOINTS.SEND(this.testInboxId);
        try {
            return await this.MailtrapInstance.post(endpoint, data);
        }
        catch (error) {
            console.error(error);
            if (axios_1.default.isAxiosError(error)) {
                throw new MailtrapAPIError(error.message);
            }
            throw error;
        }
    }
}
exports.MailtrapSDK = MailtrapSDK;
